lib/neo4j/active_node/dependent/association_methods.rb

```diff
@@ -28,7 +28,8 @@ def dependent_delete_callback(object)
         end
 
         def dependent_delete_orphans_callback(object)
-          object.as(:self).unique_nodes(self, :self, :n, :other_rel).query.delete(:n, :other_rel).exec
+          unique_query = object.as(:self).unique_nodes(self, :self, :n, :other_rel)
+          unique_query.query.detach_delete(:n).exec if unique_query
```

Comment:
Yes I think so, because even in the event where the `unique_query` is true and there are no orphan nodes to delete, executing the query will simply not match any orphan nodes (none will match `:n`) and thus not delete any.